SimpleWAWS/Code/BackgroundQueueManager.cs

```diff
@@ -191,7 +233,20 @@ private void MaintainResourceGroupLists()
 
             foreach (var resource in resources)
                 DeleteResourceGroup(resource);
-            //TODO:Add a way replenish leaking resourcegroups
+
+            // delete duplicate/leaking resource groups
+            var duplicateResourceGroups = this.FreeResourceGroups
+            .Where(rg => (rg.IsSimpleWAWSResourceName))
+            .GroupBy(s => s.GeoRegion)
+            .SelectMany(g => g.Skip(1))
+            .Union(
+             this.FreeJenkinsResourceGroups
+            .Where(rg => (rg.IsSimpleWAWSResourceName))
+            .GroupBy(s => s.GeoRegion)
+            .SelectMany(g => g.Skip(SimpleSettings.JenkinsResourceGroupsPerRegion)));
+
+            foreach (var resource in duplicateResourceGroups)
+                DeleteDuplicateResourceGroup(resource);
```

Comment:
the resource Group needs to be removed from the list of free available resources **before** it's deleted, not after. Otherwise we could start allocating it to a user while it's being deleted.